gherkin8/src/main/java/io/cucumber/core/gherkin8/formatter/ProtobufFormatter.java

```diff
@@ -0,0 +1,142 @@
+package io.cucumber.core.gherkin8.formatter;
+
+import io.cucumber.gherkin.Gherkin;
+import io.cucumber.messages.Messages;
+import io.cucumber.plugin.EventListener;
+import io.cucumber.plugin.event.EventPublisher;
+import io.cucumber.plugin.event.HookTestStep;
+import io.cucumber.plugin.event.PickleStepTestStep;
+import io.cucumber.plugin.event.Status;
+import io.cucumber.plugin.event.TestCase;
+import io.cucumber.plugin.event.TestRunFinished;
+import io.cucumber.plugin.event.TestStep;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.singletonList;
+
+public class ProtobufFormatter implements EventListener {
```

Comment:
Ideally the `ProtobufFormatter` would be living in the core module. But can't because gherking 5 and 8 have clashing artifact names. Do we want to resolve this?

Reply:
I think it’s pragmatic to keep it here for now. We can move it once we get rid of gherkin 5.

Reply:
I've resolved it by moving the protobuf formatter to it's own module. It only has a dependency on `cucumber-plugin` and a shaded dependency on Gherkin 8.